packages/router-core/src/router.ts

```diff
@@ -1688,7 +1691,7 @@ export class RouterCore<
       return {
         publicHref:
           rewrittenUrl.pathname + rewrittenUrl.search + rewrittenUrl.hash,
-        href: fullPath,
+        href: rewrittenUrl.href.replace(rewrittenUrl.origin, ''),
```

Comment:
i dont think this is right

`href` is the internal router's representation while `publicHref` contains the external one
we need to keep both worlds

cc @tannerlinsley

Reply:
Yeah, pretty sure we want to keep origin here, or at least in one of them.